golem-worker-executor-base/src/durable_host/grpc/mod.rs

```diff
@@ -0,0 +1,594 @@
+// Copyright 2024-2025 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod client;
+use tokio_stream::wrappers::UnboundedReceiverStream;
+
+use crate::durable_host::{DurabilityHost, DurableWorkerCtx};
+
+use crate::workerctx::{InvocationContextManagement, WorkerCtx};
+use anyhow::anyhow;
+use async_trait::async_trait;
+use client::*;
+use golem_common::model::invocation_context::{AttributeValue, InvocationContextSpan, SpanId};
+use golem_common::model::IdempotencyKey;
+use golem_grpc::golem_grpc_0_1_x::types::GrpcConfiguration;
+use golem_grpc::golem_grpc_0_1_x::types::GrpcMetadata;
+use golem_grpc::GrpcEntry;
+use golem_grpc::GrpcEntryPayload;
+use golem_grpc::HostGrpc;
+use prost_reflect::MethodDescriptor;
+use prost_reflect::{DescriptorPool, SerializeOptions};
+use prost_reflect::{DynamicMessage, MessageDescriptor};
+use serde::Serialize;
+use serde_json::Deserializer;
+use serde_json::{json, Value as JsonValue};
+use std::fmt::Debug;
+use std::str::FromStr;
+use std::sync::Arc;
+use tonic::metadata::MetadataMap;
+use wasmtime::component::Resource;
+use wasmtime_wasi::WasiView;
+
+async fn handle_invoke<Ctx: WorkerCtx>(
```

Comment:
This is the place where durability for gRPC calls has to be added. An instance of `Durability` has to be created, and based on the liveness check either the gRPC call's result has to be persisted, or in replay mode, the persisted result has to be returned instead of making the call. There are many examples of this in the `durable_host` module.

Reply:
ok.